File: src/main/java/io/github/mityavasilyev/springvertxreactbarapp/model/Ingredient.java
package io.github.mityavasilyev.springvertxreactbarapp.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * Describes ingredient that links to a source product
 */
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Ingredient {

    private Long id;
    private String name;
    private String description;
    private Product sourceProduct;
    private Double amount;
    private Unit unit;

}

File: src/frontend/src/model/ingredient.ts
import Product from "./product";
import Unit from "./unit";

export default class Ingredient {
  id: number;
  name: string;
  description: string;
  sourceProduct: Product;
  amount: number;
  unit: Unit;

  constructor(
    id: number,
    name: string,
    description: string,
    sourceProduct: Product,
    amount: number,
    unit: Unit
  ) {
    this.id = id;
    this.name = name;
    this.description = description;
    this.sourceProduct = sourceProduct;
    this.amount = amount;
    this.unit = unit;
  }
}

export interface IngredientDTO {
  id: number;
  name: string;
  description: string;
  sourceProduct: Product;
  amount: number;
  unit: Unit;
}